admin/js/swonpage-navigation.js

```diff
@@ -173,24 +117,55 @@
             var list = arguments[0];
             var elements = [];
             elements.push('<b style="display:block;padding:5px;padding-bottom:10px;padding-top:30px">Choose page layout to use:</b>');
+
             for (var i = 0; i < list.length; i++) {
                 var isPreview = list[i].path.indexOf('.jpg') > 0 || list[i].path.indexOf('.jpeg') > 0 || list[i].path.indexOf('.png') > 0 || list[i].path.indexOf('.gif') > 0;
-                if (isPreview) {
-                    var name = list[i].name.replace('.jpg', '').replace('.jpeg', '').replace('.png', '').replace('.gif', '');
-                    var path = list[i].path.replace('.jpg', '').replace('.jpeg', '').replace('.png', '').replace('.gif', '') + '.html';
-                    elements.push('<div class="sw-onpage-navigation-createpage-template" data-sw-onpage-createpage-template="' + path + '" style="margin:5px;padding:1px;width:250px;display:inline-block;background-color:#2F5575;color:#fff;vertical-align:top;border-radius:6px;"><b style="display:block;padding:4px">' + name + '</b><img src="' + list[i].path + '" width="100%" style="cursor:pointer;background: url(https://placehold.it/300x300)" /></div>');
+                var isLayout = list[i].path.indexOf('.html') > 0 || list[i].path.indexOf('.htm') > 0;
 
+                if (isLayout) {
+                    var name = list[i].name.replace('.html', '').replace('.htm', '');
+                    var path = list[i].path;
```

Comment:
Since path is used a "thousand" times already, maybe move this a couple of lines up.